plugins/module_utils/network/sonic/config/ip_neighbor_interfaces/ip_neighbor_interfaces.py

```diff
@@ -0,0 +1,369 @@
+                        del_commands.append(del_command)
+                        del_requests.extend(self.get_delete_ip_neighbor_interfaces_requests(del_command, have_conf))
+
+        if del_commands:
+            commands = update_states(del_commands, 'deleted')
+            new_have = self.get_new_config(commands, have)
+            requests = del_requests
+        else:
+            new_have = have
+
+        add_commands = get_diff(want, new_have, TEST_KEYS)
+        if add_commands:
+            commands.extend(update_states(add_commands, state))
+            requests.extend(self.get_modify_ip_neighbor_interfaces_requests(add_commands))
+
+        return commands, requests
+
+    def _state_merged(self, want, have):
+        """ The command generator when state is merged
+
+        :rtype: A list
+        :returns: the commands necessary to merge the provided into
+                  the current configuration
+        """
+        commands = get_diff(want, have, TEST_KEYS)
+        requests = self.get_modify_ip_neighbor_interfaces_requests(commands)
+        if commands and len(requests) > 0:
+            commands = update_states(commands, 'merged')
+        else:
+            commands = []
+
+        return commands, requests
+
+    def _state_deleted(self, want, have):
+        """ The command generator when state is deleted
+
+        :rtype: A list
+        :returns: the commands necessary to remove the current configuration
+                  of the provided objects
+        """
+        commands, requests = [], []
+        if not have:
+            return commands, requests
+        elif not want:
+            for have_conf in have:
+                commands.append({'name': have_conf['name']})
+                requests.extend(self.get_delete_ip_neighbor_interfaces_requests(have_conf, have_conf, True))
+        else:
+            for conf in want:
+                intf_name = conf['name']
+                have_conf = search_obj_in_list(intf_name, have, 'name')
+                if not have_conf:
+                    continue
+
+                # Delete all neighbor config if only interface name is specified
+                if len(conf.keys()) == 1:
+                    commands.append(conf)
+                    requests.extend(self.get_delete_ip_neighbor_interfaces_requests(conf, have_conf, True))
+                    continue
+
+                command = {}
+                for option in ('ipv4_neighbors', 'ipv6_neighbors'):
+                    if conf.get(option) and have_conf.get(option):
+                        conf_neighbors = {neighbor['ip'] for neighbor in conf[option]}
+                        have_neighbors = {neighbor['ip'] for neighbor in have_conf[option]}
+                        del_neighbors = conf_neighbors.intersection(have_neighbors)
+                        if del_neighbors:
+                            command[option] = [{'ip': neighbor} for neighbor in del_neighbors]
+
```

Comment:
Is there a reason for not supporting deletion of the (optional) mac (link-layer) address for a neighbor?